zorg/buildbot/process/factory.py

```diff
@@ -73,13 +73,25 @@ def __init__(self, steps=None, depends_on_projects=None, hint=None, **kwargs):
                 # Let's just use the given list, no need to discover.
                 self.enable_runtimes = frozenset(enable_runtimes)
 
-            # Update the list of dependencies.
-            if depends_on_projects is None:
-                self.depends_on_projects.update(self.enable_runtimes)
-
         # Build the list of projects to enable.
-        self.enable_projects = \
-            self.depends_on_projects.difference(self.enable_runtimes)
+        enable_projects = kwargs.pop('enable_projects', None)
+        if enable_projects is None or enable_projects == "auto":
+            # Assume that all non-runtimes depends_on_projects dirs are projects.
+            self.enable_projects = \
+                self.depends_on_projects.difference(self.enable_runtimes)
+        elif enable_projects == "all":
+            raise Exception("enable_projects='all' not yet supported")
```

Comment:
Did you check whether any bot attempts to use all? A quick search in the repo should be enough.

Raising an error is ok if it's unusual but we don't want to break an existing bot.

Reply:
Since I added that function argument, there cannot be any existing callers that use that argument. If there was, the buildbot would fail to startup giving that exception as error message, which I did not observe.